pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -403,6 +403,10 @@ func (f *ring1Factory) AttachablePodForObject(object runtime.Object, timeout tim
 			return nil, fmt.Errorf("invalid label selector: %v", err)
 		}
 
+	case *api.Service:
+		namespace = t.Namespace
+		selector = labels.SelectorFromSet(t.Spec.Selector)
```

Comment:
Probably only applies to services with selectors… not all services have them

Reply:
Tested with a service with no selector, and it appears the method selects all pods and returns the first one.   That is probably not ideal.

I think the right thing to do here might be to return an error when t.Spec.Selector is empty/nil., to indicate to user that the selection yields no result.  Do you see issues with this approach?

Reply:
> I think the right thing to do here might be to return an error when t.Spec.Selector is empty/nil., to indicate to user that the selection yields no result. Do you see issues with this approach?

That sounds right, though it seems like we should defer forwarding to a service at all until we get the right port selection logic

Reply:
I disagree, based on my own experience it is already a big benefit to not look up a pod before port-forward.

However, I might be biased because in our deployments service port always equals to targetPort so I may not feel the pain.